### PR TITLE
BUG: Fix user-facing AssertionError with to_html (#25608)

### LINKED ISSUE #25608: BUG: User-facing AssertionError with DataFrame.to_html(classes=<invalid type>)
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
pd.DataFrame().to_html(classes=True)
```
#### Problem description

```python-traceback
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\simon\OneDrive\code\pandas-simonjayhawkins\pandas\core\frame.py", line 2212, in to_html
    formatter.to_html(classes=classes, notebook=notebook, border=border)
  File "C:\Users\simon\OneDrive\code\pandas-simonjayhawkins\pandas\io\formats\format.py", line 729, in to_html
    html = Klass(self, classes=classes, border=border).render()
  File "C:\Users\simon\OneDrive\code\pandas-simonjayhawkins\pandas\io\formats\html.py", line 146, in render
    self._write_table()
  File "C:\Users\simon\OneDrive\code\pandas-simonjayhawkins\pandas\io\formats\html.py", line 167, in _write_table
    .format(typ=type(self.classes)))
AssertionError: classes must be list or tuple, not <class 'bool'>
```

#### Expected Output

```python-traceback
TypeError: classes must be a string, list or tuple, not <class 'bool'>
```
#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

</details>

### PR DESCRIPTION
**Pull request by ArtificialQualia**
- [x] closes #25608
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

I did not use an oxford comma in the error description, which is how it was written in the issue's expected result.

### CODE REVIEW

File: pandas/io/formats/html.py [hunk 1]
```diff
@@ -163,8 +163,8 @@ def _write_table(self, indent=0):
             if isinstance(self.classes, str):
                 self.classes = self.classes.split()
             if not isinstance(self.classes, (list, tuple)):
-                raise AssertionError('classes must be list or tuple, not {typ}'
-                                     .format(typ=type(self.classes)))
+                raise TypeError('classes must be a string, list or tuple, '
+                                'not {typ}'.format(typ=type(self.classes)))
```
Comment (gfyoung) [inline pandas/io/formats/html.py]: In general, we do use the Oxford comma in this codebase. 😉

File: doc/source/whatsnew/v0.24.2.rst [hunk 2]
```diff
@@ -64,7 +64,7 @@ Bug Fixes
 - Bug in reading a HDF5 table-format ``DataFrame`` created in Python 2, in Python 3 (:issue:`24925`)
 - Bug in reading a JSON with ``orient='table'`` generated by :meth:`DataFrame.to_json` with ``index=False`` (:issue:`25170`)
 - Bug where float indexes could have misaligned values when printing (:issue:`25061`)
--
```
Comment (gfyoung) [inline doc/source/whatsnew/v0.24.2.rst]: Move to 0.25.0.

**Review by gfyoung — APPROVED** [review]
Nice!

cc @jreback

File: doc/source/whatsnew/v0.25.0.rst [hunk 3]
```diff
@@ -214,7 +214,7 @@ I/O
 - Bug in :func:`read_json` for ``orient='table'`` when it tries to infer dtypes by default, which is not applicable as dtypes are already defined in the JSON schema (:issue:`21345`)
 - Bug in :func:`read_json` for ``orient='table'`` and float index, as it infers index dtype by default, which is not applicable because index dtype is already defined in the JSON schema (:issue:`25433`)
 - Bug in :func:`read_json` for ``orient='table'`` and string of float column names, as it makes a column name type conversion to Timestamp, which is not applicable because column names are already defined in the JSON schema (:issue:`25435`)
--
+- Bug in converting to HTML when using an invalid type for ``classes`` parameter raised ``AsseertionError`` instead of ``TypeError`` (:issue:`25608`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: an you reference ``:meth:DataFrame.to_html``; this is not a bug, rather a change.

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ArtificialQualia

**Pull request merged**